### PR TITLE
Testing for SBT build

### LINKED ISSUE #273: Run tests with SBT instead of Maven
Currently in this repo we're running tests via Maven:

http://spark-k8s-jenkins.pepperdata.org:8080/job/PR-spark-k8s-unit-tests/
`./build/mvn clean test -Pmesos -Pyarn -Phadoop-2.7 -Pkubernetes -pl core,resource-managers/kubernetes/core -am -Dtest=none -Dsuffixes='^org\.apache\.spark\.(?!SortShuffleSuite$|rdd\.LocalCheckpointSuite$|deploy\.SparkSubmitSuite$|deploy\.StandaloneDynamicAllocationSuite$).*'`

whereas in Apache Spark the tests are run via SBT:

https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/
https://github.com/apache/spark/blob/bbd163d589e7503c5cb150d934e7565b18a908f2/dev/run-tests.py#L527
`[info] Running Spark tests using SBT with these arguments:  -Phadoop-2.6 -Phive -Pyarn -Pmesos -Phive-thriftserver -Pkinesis-asl -Dtest.exclude.tags=org.apache.spark.tags.ExtendedYarnTest test`

There are subtle differences between SBT and Maven in how test are run (largely around dependency resolution) so for maximal compatibility with Apache we should be running with SBT.

This has been causing problems in the Palantir Spark repo where we cherry pick these commits into.

### PR DESCRIPTION
**Pull request by ash211**
Fixes #273

### CODE REVIEW

File: build/mvn [hunk 1]
```diff
@@ -38,7 +38,7 @@ install_app() {
   # setup `curl` and `wget` silent options if we're running on Jenkins
   local curl_opts="-L"
   local wget_opts=""
-  if [ -n "$AMPLAB_JENKINS" ]; then
+  if [ -n "$AMPLAB_JENKINS" -o -n "$K8S_JENKINS" ]; then
```
Comment (lins05) [inline build/mvn:41]: Maybe we should stick to AMPLAB_JENKINS so it would be easier getting merged in the future?

Comment (ash211) [reply build/mvn:41]: That has a bunch of other side effects that I'm not entirely sure we want -- definitely open to trimming this down before merging.

I'm still experimenting to try to get http://spark-k8s-jenkins.pepperdata.org:8080/job/PR-spark-k8s-unit-tests-SBT-TESTING working

**Comment by foxish** [comment]
It seems like the unit test failure here could be a flake?

**Comment by ash211** [comment]
@foxish it doesn't look like the kind of failure that would be a flake to me:

```
========================================================================
Running PySpark tests
========================================================================
No handlers could be found for logger "root"
Running PySpark tests. Output is in /home/jenkins/workspace/PR-spark-k8s-unit-tests-SBT-TESTING/python/unit-tests.log
Will test against the following Python executables: ['python']
Will test the following Python modules: ['pyspark-core', 'pyspark-ml', 'pyspark-mllib', 'pyspark-sql', 'pyspark-streaming']
/bin/python: No module named numpy
```

I think the python install on this box doesn't have numpy

**Comment by ash211** [comment]
The most recent test failed due to:

```
[info] KafkaSourceStressSuite:
[error] running /home/jenkins/workspace/PR-spark-k8s-unit-tests-SBT-TESTING/build/sbt -Phadoop-2.7 -Phive -Pyarn -Pmesos -Phive-thriftserver -Pkinesis-asl test ; received return code 143
```

so I think the Kafka tests might have caused a system OOM (143-128 = 15 which is SIGTERM).  @kimoonkim do you have any metrics on this box to determine if its physical memory was fully committed?

**Comment by ash211** [comment]
rerun unit tests SBT please

**Comment by mccheah** [comment]
This is probably stale now. Closing in favor of #344.

**Pull request closed without merging**